src/guides/metamask-api/index.md

```diff
@@ -0,0 +1,1094 @@
+
+Update `pages/index/tsx`
+
+```typescript
+import type { NextPage } from "next";
+import { useEffect } from "react";
+import Wallet from "../components/Wallet";
+import { useListen } from "../hooks/useListen";
+import { useMetamask } from "../hooks/useMetamask";
+
+const Home: NextPage = () => {
+  const { dispatch } = useMetamask();
+  const listen = useListen();
+
+  useEffect(() => {
+    if (typeof window !== undefined) {
+      // start by checking if window.ethereum is present, indicating a wallet extension
+      const ethereumProviderInjected = typeof window.ethereum !== "undefined";
+      // this could be other wallets so we can verify if we are dealing with metamask
+      // using the boolean constructor to be explecit and not let this be used as a falsy value (optional)
+      const isMetamaskInstalled =
+        ethereumProviderInjected && Boolean(window.ethereum.isMetaMask);
+
+      const local = window.localStorage.getItem("metamaskState");
+
+      // user was previously connected, start listening to MM
+      if (local) {
+        listen();
+      }
+
+      // local could be null if not present in LocalStorage
+      const { wallet, balance } = local
+        ? JSON.parse(local)
+        : // backup if local storage is empty
+          { wallet: null, balance: null };
+
+      dispatch({ type: "pageLoaded", isMetamaskInstalled, wallet, balance });
+    }
+  }, []);
+
+  return (
+    <>
+      <Wallet />
+    </>
+  );
+};
+
+export default Home;
+```
+
+In this last page update to `pages/index.tsx` we have relocated the `useEffect` from the `hooks/useMetaMask.tsx` page for separation of concerns since the hook is consuming **dispatch** as a proper next step would to be creating a layout page with NextJS, but since we only have one page we simply added this code here.
+
+We have updated our `hooks/useMetamask.tsx` page's **PageLoadAction** to include **wallet** and **balance** as well as the code required to access our local storage and rehydrate our app.
+
+With those changes in place we have also updated our `components/Wallet.tsx` page to use our `useListen` hook since we are using that code in multiple places now, updated the `showAddToken` variable to a more descriptive name of `isConnected`, and added a `handleDisconnect()` function to dispatch an action clearing local storage in our browser.
+
+This also required a slight update to our JSX/HTML to display our buttons more neatly.
+
+[Checkout the Diff to see what changed](https://github.com/GuiBibeau/web3-unleashed-demo/pull/4/files)
+
+## Remove Listeners after Disconnect
+
+We have one final change we want to make to ensure that we stop listening to changes once the user has disconnected their wallet.
+
+We will update the `` and `` files. This will make TypeScript definitions file aware of the `removeAllListeners()` method we will be using as well add necessary code to the `disconnect` case inside the `metamaskReducer`.
```

Comment:
```suggestion
We will update the `` and `` files. This will make TypeScript definitions file aware of the `removeAllListeners()` method we will be using, as well add necessary code to the `disconnect` case inside the `metamaskReducer`.
```